test/unittests/eof_validation_test.cpp

```diff
@@ -1211,3 +1211,198 @@ TEST_F(eof_validation, max_nested_containers)
 
     add_test_case(code, EOFValidationError::success);
 }
+
+// Summary of validity of combinations of referencing instructions with instructions inside
+// referenced containers.
+// Rows are instructions referencing subcontainers or rules for top-level container.
+// Columns are instructions inside referenced subcontainer or none if it ends with REVERT/INVALID.
+//
+//    |                | STOP   | RETURN | RETURNCONTRACT | none |
```

Comment:
I presume you're omitting `TXCREATE` because it is getting removed? Otherwise would be same as `EOFCREATE`, correct?

Also, to anticipate the creation txs, I'm wondering whether it has same row as `EOFCREATE` or all terminating opcodes are allowed there? :thinking: A creation tx which deploys a bunch of contracts using `EOFCREATE`s, but doesn't `RETURNCONTRACT` itself seems valid. Or just runs some logic and doesn't even create any contract.

Actually... hooold on, what I'm saying is not according to spec of EIP-7698, but maybe sensible anyway? Especially, if we flip the requirement that creation tx top level is initcode-mode, and have it in runtime-mode, **requiring** an `EOFCREATE` inside to actually create something?

I need to look at my creation tx PR to ponder whether it's cleaner or not...